w1/models/cat_model.go

```diff
@@ -385,3 +385,25 @@ func GetCats(opts GetCatOption, userId string) ([]CatOut, error) {
 
 	return cats, nil
 }
+
+func UpdateHasMatchedCat(catIds []string, userIds []string, db *sqlx.DB) error {
+	arg := map[string]interface{}{
+		"catIds":  catIds,
+		"userIds": userIds,
+	}
+	query := `UPDATE cats SET has_matched = true WHERE id IN (:catIds) AND owner_id IN (:userIds)`
+	query, args, err := sqlx.Named(query, arg)
+	if err != nil {
+		return err
+	}
+
+	query, args, err = sqlx.In(query, args...)
+	if err != nil {
+		return err
+	}
+
+	query = db.Rebind(query)
+	_, err = db.Exec(query, args...)
+
+	return err
+}
```

Comment:
The responsibility of this function is just for updating cat's has_matched status, isn't it ? Since it's not used directly by the client, may be it's still okay to delete change the has_matched status only by cat's id. Why ? i guess before calling this function, we have already validate in other place.

Reply:
yes it is, i think so too before, but somehow ended up choose it this way to be more safe.